Tests/FoundationTests/NSAttributedStringExtensionsTests.swift

```diff
@@ -319,6 +319,47 @@ final class NSAttributedStringExtensionsTests: XCTestCase {
         XCTAssertEqual(string1.string, "Test Appending")
         #endif
     }
+    
+    func testJoinedWithSeparator() {
+        let string1 = NSAttributedString(
+            string: "Hello",
+            attributes: [
+                .strokeWidth: NSNumber(value: 1),
+                .kern: NSNumber(value: 2)
+            ]
+        )
+        let string2 = NSAttributedString(
+            string: " ",
+            attributes: [
+                .expansion: NSNumber(value: 3),
+                .obliqueness: NSNumber(value: 4)
+            ]
+        )
+        let string3 = NSAttributedString(string: "World", attributes: [:])
+        let expectation = NSMutableAttributedString(string: "Hello World", attributes: [:])
+        expectation.addAttribute(
+            .strokeWidth,
+            value: NSNumber(value: 1),
+            range: .init(location: 0, length: "Hello".count)
+        )
+        expectation.addAttribute(
+            .kern,
+            value: NSNumber(value: 2),
+            range: .init(location: 0, length: "Hello".count)
+        )
+        expectation.addAttribute(
+            .expansion,
+            value: NSNumber(value: 3),
+            range: .init(location: "Hello".count, length: " ".count)
+        )
+        expectation.addAttribute(
+            .obliqueness,
+            value: NSNumber(value: 4),
+            range: .init(location: "Hello".count, length: " ".count)
+        )
+        XCTAssertEqual([string1, string2, string3].joined(separator: ""), expectation)
+        XCTAssertEqual([].joined(separator: NSAttributedString(string: "Hello")), NSAttributedString(string: ""))
```

Comment:
Can you add tests for non-empty string separator, and an empty attributed string separator?